src/Utils.hs

```diff
@@ -0,0 +1,15 @@
+module Utils (dosedLists, saltedRange) where
+
+
+-- dosedLists 2 [1..6] = [[1,2],[3,4],[5,6]]
+-- dosedLists 3 [1..6] = [[1,2,3],[4,5,6]]
+dosedLists :: (Num a) => Int -> [a] -> [[a]]
+dosedLists doseSize list = dose list []
+  where dose list result
+          | length taken < doseSize = result
+          | otherwise = taken : dose (drop doseSize list) result
+            where taken = take doseSize list
+
+
+saltedRange :: Float -> (Float, Float) -> Float
+saltedRange salt (from, to) = salt * (to - from) + from
```

Comment:
I look at this function and I don't understand what it does and what it's intended for. Could you please elaborate on that?

You also moved it to a utility module, which means that you believe that this function is going to be highly reusable accross the entire code base. Could you please elaborate on how you see such a specific function to be so highly reusable?